### PR TITLE
internal/namingschema: simplify the namingschema

### PR DESCRIPTION
**Pull request by knusbaum**
### What does this PR do?

This commit refactors the namingschema interface for service names
and operation names for performance and clarity reasons.

### Motivation

We have discovered a performance degradation recently that was tied in part to
the addition of namingschema to the net/http tracing. This was hard to track down.

This change both fixes the performance degradation, eliminates other possible undiscovered
performance issues around namingschema, and clarifies the code so it is easier
to tell what is happening. 

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: internal/namingschema/op.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	case SchemaV1:
+		op := "request"
+		if isMessagingSendOp(service, operation) {
+			op = "send"
+		}
+		return fmt.Sprintf("aws.%s.%s", strings.ToLower(service), op)
+	default:
+		return overrideV0
+	}
+}
```
Comment (knusbaum) [inline internal/namingschema/op.go:191]: TODO: add tests for these.

File: internal/namingschema/service_name_test.go [hunk 2]
```diff
@@ -16,49 +16,48 @@ import (
 
 func TestNewDefaultServiceName(t *testing.T) {
```
Comment (rarguelloF) [inline internal/namingschema/service_name_test.go:17]:
```suggestion
func TestServiceName(t *testing.T) {
```

File: internal/namingschema/op_test.go [hunk 3]
```diff
@@ -14,157 +14,181 @@ import (
 )
 
 func TestOpContribSchemas(t *testing.T) {
```
Comment (rarguelloF) [inline internal/namingschema/op_test.go:16]:
```suggestion
func TestOpName(t *testing.T) {
```

File: internal/namingschema/op_test.go [hunk 4]
```diff
@@ -14,157 +14,181 @@ import (
...
-	optOverrideV0 := namingschema.WithOverrideV0("override-v0")
+	optOverrideV0 := "override-v0"
 
 	testCases := []struct {
 		name      string
-		newSchema func() *namingschema.Schema
+		newSchema func() string
```
Comment (rarguelloF) [inline internal/namingschema/op_test.go:21]:
```suggestion
		getOpName func() string
```

File: internal/namingschema/service_name_test.go [hunk 5]
```diff
@@ -16,49 +16,48 @@ import (
...
 	defaultServiceName := "default"
-	optOverrideV0 := namingschema.WithOverrideV0("override-v0")
 
 	testCases := []struct {
 		name          string
 		schemaVersion namingschema.Version
 		ddService     string
 		setup         func() func()
-		opts          []namingschema.Option
+		call          func() string
```
Comment (rarguelloF) [inline internal/namingschema/service_name_test.go:25]:
```suggestion
		getServiceName          func() string
```

**Review by rarguelloF — COMMENTED** [review]
Nice work 🎉  I left some small nits for some some files that I missed in the previous review.

I'm also thinking the constant name `defaultServiceName` that is in most contrib packages is now kind of confusing, since in v1 is no longer the default, just the fallback when DD_SERVICE is not set... Anyway, no need to tackle in this PR, but something to consider for a future refactor.

**Review by rarguelloF — APPROVED** [review]

**Review by zarirhamza — APPROVED** [review]

**Pull request merged**